File: client/qr-code-app/qr-code-app.py
import re
from enum import Enum
import numpy as np

def QrCodeGenerator(str):
    symbolType = ['0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z',' ','$','%','*','+','-','.','/',':']
    MULTIPLIER_CONST = 45

    def strToChunks(str):
        chunkSize = 2
        res = re.findall('.{1,2}', str)
        return [[i] for i in res]

    def charToSymbolTypeChunks(arr):
        res = []
        el = []
        x = np.array(symbolType)
        for a in arr:
            for b in a:
                y = np.array(b)
                index = np.argsort(y)
                sorted_y = y[index]
                sorted_index = np.searchsorted(sorted_y, y)

                yindex = np.take(index, sorted_index, mode="clip")
                mask = x[yindex] != y

                result = np.ma.array(yindex, mask=mask)
                print(result)
                


                # for c in b:
                    # if d == c:
                        # print(symbolType.index(d))
                            # result = any(elem in symbolType for elem in c)
                    # if (result):

                    #     print(result)
                        # print(d)
                        # if c.upper() == d:
                            #    el.append(d.index(c.upper()))
            # res.append(el)
                # if SymbolType.index(first):
                #     print(SymbolType.index(first))
                # if SymbolType.index(second):
                #     print(SymbolType.index(second))
                # else:
                #     print('didnt exist')
        # print(res)

    def resFromPairsToArr():
        return

    def toBinary(setOfNums):
        res = None

    def isEq(self, other):
        return self.Value == other.Value
    
    res = strToChunks(str)
    charToSymbolTypeChunks(res)
    # print(res)

str = 'lorem ipsum $ goolge me'
QrCodeGenerator(str)

File: client/qr-code-app/src/app/qr-code-app.ts
export const symbolType = ['0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z',' ','$','%','*','+','-','.','/',':']
class QrCodeGenerator {
    public dataToParse:string;

    constructor(data:string) {
        this.dataToParse = data;
        this.dataToPairs(data);
    }

    dataToPairs(data:string):string {
        let res:any[] = [];
        let arr = data.toUpperCase().match(/.{1,2}/g);
        
        for (let str of arr) {
            let el = str.split('');
            if (el) {
                let a = el[0];
                let b = el[1];
                let first = symbolType.findIndex(val => val == a)
                console.log
                let second = symbolType.findIndex(val => val == b)
                let sum = first * 45 + second;
                let bin = (sum >>> 0).toString(2)
                res.push(bin)
            }
        }
        
        return res.join('');
    }
}

let str = 'lorem ipsum $ goolge me'
new QrCodeGenerator(str)
